[Workflows] Parallel step reference (#7691)
elastic/docs-content: explore-analyze/workflows/steps/flow-control-steps.md

| Iterate over an array | [`foreach`](#foreach) |
| Loop until a condition is false | [`while`](#while) |
| Multi-way dispatch on a value | [`switch`](#switch) |
| Run independent work at the same time | [`parallel`](#parallel) {applies_to}`stack: preview 9.5+` {applies_to}`serverless: preview` |
| Pause for a fixed duration | [`wait`](#wait) |
| Exit a loop early | [`loop.break`](#loop-break) |
| Skip to the next loop iteration | [`loop.continue`](#loop-continue) |
| Pause for human input (human-in-the-loop) | [`waitForInput`](#waitforinput) |
| Pause for approve/reject (human-in-the-loop) | [`waitForApproval`](/explore-analyze/workflows/steps/wait-for-approval.md) {applies_to}`stack: preview 9.5+` {applies_to}`serverless: preview` |

> Two step types fan work out. Use [`parallel`](#parallel) when the work runs inside the current workflow and you need the results, and [`workflow.executeAsync`](/explore-analyze/workflows/steps/composition.md#workflow-executeasync) when each unit of work should run as its own independent execution that the parent doesn't wait for.

Reviewer: Maybe I lack context, but I just don't understand what you're trying to say with that first sentence 😄

Author: Agree it is a bit vague – I'll rewrite.
```suggestion
Use [`parallel`](#parallel) to run independent work inside the current workflow and collect the results. For fan-out across separate workflow executions that the parent doesn't wait for, use [`workflow.executeAsync`](/explore-analyze/workflows/steps/composition.md#workflow-executeasync).

```

For the full parameter reference, refer to [Switch step](/explore-analyze/workflows/steps/switch.md).

## `parallel` [parallel]

```{applies_to}
stack: preview 9.5+
serverless: preview
```

> Run branches at the same time and continue when all of them reach a terminal state. Choose one of two modes: dynamic fan-out (`foreach` plus `steps`) runs the same body once per item in a runtime list, and static branches (`branches`) run a fixed set of named branches that each do different work. A step can't use both.

Reviewer 2: **Nit / Vale:** "all of them" → prefer "every branch" (matches the wording already used in `parallel.md`).

Suggested:
```markdown
Run branches at the same time and continue when every branch reaches a terminal state.
```

Author: Resolved in [29041cc](https://github.com/elastic/docs-content/pull/7691/commits/29041ccc7e1de107ff411c7d02c7ee6a6afc34af).
